index.js

```diff
@@ -4,18 +4,24 @@ import fetch from "./doppler.js";
 // For local testing
 if (process.env.NODE_ENV === "development" && process.env.DOPPLER_TOKEN) {
   process.env["INPUT_DOPPLER-TOKEN"] = process.env.DOPPLER_TOKEN;
+  process.env["INPUT_DOPPLER-PROJECT"] = process.env.DOPPLER_PROJECT;
+  process.env["INPUT_DOPPLER-CONFIG"] = process.env.DOPPLER_CONFIG;
 }
 
-const DOPPLER_META = [
-  "DOPPLER_PROJECT",
-  "DOPPLER_CONFIG",
-  "DOPPLER_ENVIRONMENT",
-];
+const DOPPLER_META = ["DOPPLER_PROJECT", "DOPPLER_CONFIG", "DOPPLER_ENVIRONMENT"];
 
 const DOPPLER_TOKEN = core.getInput("doppler-token", { required: true });
 core.setSecret(DOPPLER_TOKEN);
 
-const secrets = await fetch(DOPPLER_TOKEN);
+// Project and Config are required if a Personal token supplied
+const IS_PERSONAL_TOKEN = DOPPLER_TOKEN.indexOf("dp.pt") >= 0;
```

Comment:
We can make this check a bit more precise
```suggestion
const IS_PERSONAL_TOKEN = DOPPLER_TOKEN.startsWith("dp.pt.");
```